runtime-modules/content/src/nft/mod.rs

```diff
@@ -382,15 +382,23 @@ impl<T: Trait> Module<T> {
             .with_member_owner(winner_id)
     }
 
-    pub(crate) fn ensure_owner_account_id(
+    /// NFT owned by:
+    /// - Member: member controller account is used
+    /// - Channel: then if reward account is:
+    ///    - `Some(acc)` -> use `acc` as reward account
+    ///    - `None` -> then if channel owner is:
+    ///      - `Member` -> use member controller account
+    ///      - `CuratorGroup` -> Error
+    pub(crate) fn ensure_nft_owner_account_id(
         channel_id: T::ChannelId,
         nft: &Nft<T>,
     ) -> Result<T::AccountId, DispatchError> {
         match nft.owner {
             NftOwner::Member(member_id) => T::MemberAuthenticator::controller_account_id(member_id),
-            NftOwner::ChannelOwner => Self::channel_by_id(channel_id)
-                .reward_account
-                .ok_or_else(|| Error::<T>::RewardAccountIsNotSet.into()),
+            NftOwner::ChannelOwner => {
+                let channel = Self::ensure_channel_exists(&channel_id)?;
+                Self::ensure_reward_account(&channel)
```

Comment:
Can we rename `ensure_reward_account` to `ensure_channel_has_beneficiary_account`.

Reply:
addressed in 8d223f3594daea61e53289ee82f1080bbe8c1e9b